pkg/antctl/raw/migrate/checkenv/command.go

```diff
@@ -0,0 +1,291 @@
+			return err
+		}
+	}
+
+	networkPolicies, err := c.calicoClient.ProjectcalicoV3().NetworkPolicies("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list NetworkPolicies: %v", err)
+		return err
+	}
+	for _, np := range networkPolicies.Items {
+		if np.Spec.ServiceAccountSelector != "" {
+			return fmt.Errorf("NetworkPolicy %s has set field spec.serviceAccountSelector, which is not supported by Antrea", np.Name)
+		}
+		if err := checkCalicoRules(np.Spec.Egress, np.Name); err != nil {
+			return err
+		}
+		if err := checkCalicoRules(np.Spec.Ingress, np.Name); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkCalicoRules(rules []v3.Rule, gnpName string) error {
+	for _, rule := range rules {
+		if rule.ICMP != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.icmp or spec.ingress.icmp,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.NotICMP != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.notICMP or spec.ingress.icmp,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.NotProtocol != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.notProtocol or spec.ingress.notProtocol,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.Metadata != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.metadata or spec.ingress.metadata,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.HTTP != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.http or spec.ingress.http,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.Source.NotSelector != "" || rule.Destination.NotSelector != "" {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.notSelector or spec.ingress.notSelector,"+
+				"which is not supported by Antrea", gnpName)
+		}
+
+		if rule.Source.NotNets != nil || rule.Destination.NotNets != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.notNets or spec.ingress.notNets,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if rule.Source.NotPorts != nil || rule.Destination.NotPorts != nil {
+			return fmt.Errorf("GlobalNetworkPolicy/NetworkPolicy %s has set field spec.egress.notPorts or spec.ingress.notPorts,"+
+				"which is not supported by Antrea", gnpName)
+		}
+		if err := checkSelector(rule.Source.Selector); err != nil {
+			return err
+		}
+		if err := checkSelector(rule.Source.NamespaceSelector); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkSelector checks if the selector is supported by Antrea.
+func checkSelector(selector string) error {
+	if strings.Contains(selector, "||") {
+		return fmt.Errorf("selector %s is not supported by Antrea", selector)
```

Comment:
Keep the log format consistent, better to include the NP name in the returned log as well, Maybe in the caller.